### PR TITLE
1321 add dropdown button variant

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Add dropdown option to button, add cypress tests. Also add icPopoverClosed event to help users using the dropdown with popover menu. 

## Related issue
#1321 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

**Comment by GCHQ-Developer-530** [comment]
> There seems to be extra padding on the right... does this match the designs? Worth having @DBD324 looking into this also. Screenshot 2024-01-17 at 15 00 13

I noticed that as well. It does seem to match the designs though. @DBD324 did mention this yesterday saying that they were worried that if there was no left icon it might look too big but they left it as is. 
Screenshot 2024-01-17 at 15 06 27

File: packages/web-components/src/components/ic-button/ic-button.tsx [hunk 1]
```diff
@@ -93,6 +94,16 @@ export class Button {
    */
   @Prop() download?: string | boolean = false;
 
+  /**
+   * If `true`, the button will show a dropdown icon.
+   */
+  @Prop() dropdown?: boolean = false;
+
+  /**
+   * If `true`, the button will be blah.
```
Comment (GCHQ-Developer-299) [inline packages/web-components/src/components/ic-button/ic-button.tsx:103]: Will the button be blah?  I reckon this comment should say something else ;)

File: packages/react/cypress-visual-screenshots/baseline/IcButton.cy.tsx-dropdown.png
Comment (GCHQ-Developer-299) [inline packages/react/cypress-visual-screenshots/baseline/IcButton.cy.tsx-dropdown.png:1]: For a cypress screenshot i'd like to see maybe the dropdown example (like you've got in Storybook) open, so we can see the arrow having switched and the additional content being displayed.

**Comment by GCHQ-Developer-465** [comment]
> > There seems to be extra padding on the right... does this match the designs? Worth having @DBD324 looking into this also. Screenshot 2024-01-17 at 15 00 13
> 
> I noticed that as well. It does seem to match the designs though. @DBD324 mentioned this yesterday while having a look at it, saying that they were worried that if there was no left icon it might look too big but they left it as is.
> 
> Screenshot 2024-01-17 at 15 06 27

Nice! I still believe that the padding looks off. After some research and looking at others already available online they do not have this extra padding. For instance: 

- [Flowbite](https://www.flowbite-react.com/docs/components/dropdown#default-dropdown)
- [Core UI](https://coreui.io/react/docs/components/dropdown/)
- [React](https://react-bootstrap.netlify.app/docs/components/dropdowns/)

and so on...

This would be a question to @DBD324... has this been taken in consideration. Have you looked at the other sources out there? Why is there extra space and padding? Is this necessary as it looks out of place and should be symmetrical to the left side?

**Comment by DBD324** [comment]
> > > There seems to be extra padding on the right... does this match the designs? Worth having @DBD324 looking into this also. Screenshot 2024-01-17 at 15 00 13
> > 
> > 
> > I noticed that as well. It does seem to match the designs though. @DBD324 mentioned this yesterday while having a look at it, saying that they were worried that if there was no left icon it might look too big but they left it as is.
> > Screenshot 2024-01-17 at 15 06 27
> 
> Nice! I still believe that the padding looks off. After some research and looking at others already available online they do not have this extra padding. For instance:
> 
> * [Flowbite](https://www.flowbite-react.com/docs/components/dropdown#default-dropdown)
> * [Core UI](https://coreui.io/react/docs/components/dropdown/)
> * [React](https://react-bootstrap.netlify.app/docs/components/dropdowns/)
> 
> and so on...
> 
> This would be a question to @DBD324... has this been taken in consideration. Have you looked at the other sources out there? Why is there extra space and padding? Is this necessary as it looks out of place and should be symmetrical to the left side?

Apologies for delayed response. Originally this padding was intentional to simplify the button development and keen it consistent. The right hand padding doesn't look too imbalanced when you use a decorative icon on the button at the same time.

However, I have played around with it and tested a few different icon options and I agree we should reduce the right padding to 8px from 16px to account for the visual alignment issue caused by the small glyph within the dropdown icon. The left padding should remain at 16px though.

**Review by DBD324 — CHANGES_REQUESTED** [review]
The vertical positioning of the dropdown and drop-up icon looks a little off when compared with the Figma designs. It looks too low by a few pixels. Please could you check. 

Also, see comments about right side padding. Agree that we can reduce this to 8px to account for the visual alignment issue.

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]
Much better! 👍  Happy if @DBD324 is happy.

File: packages/web-components/src/components/ic-button/ic-button.tsx [hunk 2]
```diff
@@ -93,6 +94,16 @@ export class Button {
    */
   @Prop() download?: string | boolean = false;
 
+  /**
+   * If `true`, the button will show a dropdown icon.
+   */
+  @Prop() dropdown?: boolean = false;
+
+  /**
+   * If `true`, the button will be in an expanded state.
```
Comment (GCHQ-Developer-112) [inline packages/web-components/src/components/ic-button/ic-button.tsx:103]:
```suggestion
   * If `true`, the aria-expanded value will be set as true.
```

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-112 — APPROVED** [review]

**Review by DBD324 — APPROVED** [review]
Changes look great. The padding and icon placement look much better now. Thanks

**Pull request merged**